tools/deb/istio-iptables.sh

```diff
@@ -0,0 +1,167 @@
+
+# After more testing, the goal is to replace and unify the script in K8S - by generating
+# the sidecar image using the .deb file created by proxy.
+
+function usage() {
+  echo "${0} -p PORT -u UID [-h]"
+  echo ''
+  echo '  -p: Specify the envoy port to which redirect all TCP traffic (default $ENVOY_PORT = 150001)'
+  echo '  -u: Specify the UID of the user for which the redirection is not'
+  echo '      applied. Typically, this is the UID of the proxy container (default to uid of $ENVOY_USER, uid of istio_proxy, or 1337)'
+  echo '  -i: Comma separated list of IP ranges in CIDR form to redirect to envoy (optional)'
+  echo ''
+  echo 'Using environment variables in $ISTIO_SIDECAR_CONFIG (default: /var/lib/istio/envoy/sidecar.env)'
+}
+
+set -o nounset
+set -o pipefail
+IFS=,
+
+ISTIO_SIDECAR_CONFIG=${ISTIO_SIDECAR_CONFIG:-/var/lib/istio/envoy/sidecar.env}
+if [ -r ${ISTIO_SIDECAR_CONFIG} ]; then
+  . ${ISTIO_SIDECAR_CONFIG}
+fi
+
+IP_RANGES_INCLUDE=${ISTIO_SERVICE_CIDR:-}
+
+while getopts ":p:u:e:i:h" opt; do
+  case ${opt} in
+    p)
+      ENVOY_PORT=${OPTARG}
+      ;;
+    u)
+      ENVOY_UID=${OPTARG}
+      ;;
+    i)
+      IP_RANGES_INCLUDE=${OPTARG}
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+
+# TODO: more flexibility - maybe a whitelist of users to be captured for output instead of
+# a blacklist.
+if [ -z "${ENVOY_UID:-}" ]; then
+  # Default to the UID of ENVOY_USER and root
+  ENVOY_UID=$(id -u ${ENVOY_USER:-istio-proxy})
+  if [ ! $? == 0 ]; then
+     echo "Invalid istio user $ENVOY_UID $ENVOY_USER"
+     exit 1
+  fi
+  # If ENVOY_UID is not explicitly defined (as it would be in k8s env), we add root to the list,
+  # for ca agent.
+  ENVOY_UID=${ENVOY_UID},0
+fi
+
+# Remove the old chains, to generate new configs.
+iptables -t nat -D PREROUTING -p tcp -j ISTIO_INBOUND 2>/dev/null
+iptables -t nat -D OUTPUT -p tcp -j ISTIO_OUTPUT 2>/dev/null
+
+# Flush and delete the istio chains
+iptables -t nat -F ISTIO_OUTPUT 2>/dev/null
+iptables -t nat -X ISTIO_OUTPUT 2>/dev/null
+iptables -t nat -F ISTIO_INBOUND 2>/dev/null
+iptables -t nat -X ISTIO_INBOUND 2>/dev/null
+iptables -t nat -F ISTIO_REDIRECT 2>/dev/null
+iptables -t nat -X ISTIO_REDIRECT 2>/dev/null
+
+if [ "${1:-}" = "clean" ]; then
+  # Only cleanup, don't add new rules.
+  exit 0
+fi
+
+# Create a new chain for redirecting inbound traffic to the common Envoy port.
+# In the ISTIO_INBOUND and ISTIO_OUTBOUND chains, '-j RETURN' bypasses Envoy
+# and '-j ISTIO_REDIRECT' redirects to Envoy.
+iptables -t nat -N ISTIO_REDIRECT
+iptables -t nat -A ISTIO_REDIRECT -p tcp -j REDIRECT --to-port ${ENVOY_PORT:-15001}
+
+# Handling of inbound ports. Traffic will be redirected to Envoy, which will process and forward
+# to the local service.
+if [ -n "${ISTIO_INBOUND_PORTS:-}" ]; then
+  iptables -t nat -N ISTIO_INBOUND
```

Comment:
Whitelist of ports we capture on 'inbound' - i.e. ports services run on. 
 The .env config file has more detailed comments on all settings.

I also have the * option ( which means everything except 22 - bad things happen if we try to capture 22 ), with
an exclude list.